vittles/sensitivity_lib.py

```diff
@@ -351,31 +355,44 @@ def __init__(
 
         self.set_base_values(
             opt_par_value, hyper_par_value,
-            hessian_at_opt, cross_hess_at_opt,
+            hessian_at_opt, cross_hess_at_opt,sens_mat,
             factorize_hessian,
             validate_optimum=validate_optimum,
-            grad_tol=self._grad_tol)
+            grad_tol=self._grad_tol,
+            system_solver=system_solver,
+            compute_hess=compute_hess,
+            cg_opts=cg_opts)
 
     def set_base_values(self,
                         opt_par_value, hyper_par_value,
-                        hessian_at_opt, cross_hess_at_opt,
+                        hessian_at_opt, cross_hess_at_opt,sens_mat,
                         factorize_hessian,
-                        validate_optimum=True, grad_tol=None):
+                        validate_optimum=True, grad_tol=None,
+                        system_solver=None, compute_hess=True, cg_opts={}):
 
         # Set the values of the optimal parameters.
         self._opt0 = deepcopy(opt_par_value)
         self._hyper0 = deepcopy(hyper_par_value)
 
         # Set the values of the Hessian at the optimum.
         if hessian_at_opt is None:
-            self._hess0 = self._obj_fun_hessian(self._opt0, self._hyper0)
+            if compute_hess or (system_solver is None):
+                print('computing hessian ... ')
+                self._hess0 = self._obj_fun_hessian(self._opt0, self._hyper0)
+            else:
+                self._hess0 = None
         else:
             self._hess0 = hessian_at_opt
-        if self._hess0.shape != (len(self._opt0), len(self._opt0)):
+        if (self._hess0 is not None) and \
+            (self._hess0.shape != (len(self._opt0), len(self._opt0))):
             raise ValueError('``hessian_at_opt`` is the wrong shape.')
 
-        method = 'factorization' if factorize_hessian else 'cg'
-        self.hess_solver = SystemSolver(self._hess0, method)
+        if system_solver is None:
```

Comment:
At this point, I believe everything will fail if ``compute_hess = False``, ``hessian_at_opt`` is ``None``, and ``system_solver`` is ``None``.

To be fair, checking the consistency of all the specifications will be a mess and hard to get right.  But for that reason, I think this would be better done with two different init methods.